File: Generating/builder/python/ContactBuilder.py
from Contact import Contact

class ContactBuilder:
  def __init__(self):
    self.firstName = None
    self.lastName = None
    self.phoneNumber = None
    pass

  def setFirstName(self, name):
    self.firstName = name
    return self
  
  def setLastName(self, name):
    self.lastName = name
    return self

  def setPhone(self, phone):
    self.phoneNumber = phone
    return self
  
  def build(self):
    if self.firstName is None or self.lastName is None or self.phoneNumber is None:
      raise ValueError('Not all needed value from create')
    return Contact(self.firstName + ' ' + self.lastName, self.phoneNumber)

File: Generating/builder/src/ContactBuilder.ts
import { Contact } from './Contact'
import { IBuilder, IContact } from './interfaces'

export class ContactBuilder implements IBuilder {  
  public firstName: string
  public lastName: string
  public phoneNumber: string

  setFirstName(name: string): IBuilder {
    this.firstName = name
    return this
  }

  setLastName(name: string): IBuilder {
    this.lastName = name
    return this
  }

  setPhoneNumber(phone: string): IBuilder {
    this.phoneNumber = phone
    return this
  }

  finally(): IContact {
    if (!this.firstName || !this.lastName || !this.phoneNumber) {
      throw new Error('Not all params for create')
    }

    return new Contact({
      fullName: `${this.firstName} ${this.lastName}`,
      phoneNumber: this.phoneNumber
    })
  }
}